lib/ruby_smb/smb2/packet/session_setup_request.rb

```diff
@@ -30,6 +30,17 @@ def set_type1_blob(type1_message)
           self.buffer = gss_blob
         end
 
+        # Takes a serialized NTLM Type 1 message inserts it into the {RubySMB::SMB2::Packet::SessionSetupRequest#buffer}
+        # as well as updating the {RubySMB::SMB2::Packet::SessionSetupRequest#security_buffer_length}.
+        # This method DOES NOT wrap the NTLM Type 1 message in GSS.
+        #
+        # @param type1_message [String] the serialized NTLM Type 1 message
+        # @return [void]
+        def set_ntlm_type1_blob(type1_message)
```

Comment:
This looks like it's not necessary, the caller should be able to set the buffer directly. To remove the necessity on setting the length field correctly, you can update the field to use an `initial_value` lambda function.
```
        uint16                     :security_buffer_length,  label: 'Security Buffer Length', initial_value -> { buffer.length }
```

With this in place, you should be able to remove both the `set_ntlm_type1_blob` and `set_ntlm_type3_blob` functions. Alternatively, if we don't want to update that field, those two functions should probably be combined into a more generic one like `set_security_buffer` since the body of the functions are the same and they have nothing to do with NTLM.

Reply:
To keep things simple/flexible for the future, the previous methods have been consolidated into a single setter